Formula/xdk.rb

```diff
@@ -0,0 +1,22 @@
+class Xdk < Formula
+  desc "Ecstasy Development Kit (XDK)"
+  homepage "https://github.com/xtclang/xvm/"
+  url "https://github.com/xtclang/xvm/releases/download/v0.4.2/xdk-0.4.2.tar.gz"
+  sha256 "b0b5217abf05de986e6211a7c8665b304fa89df1726749e4f1dd7eff1cbbe1fe"
+  license "Apache-2.0"
+
+  depends_on "openjdk"
+
+  def install
+    libexec.install buildpath.children
+    osname = OS.mac? ? "macos" : OS.kernel_name.downcase
+    bin.install_symlink "#{libexec}/bin/#{osname}_launcher" => "xec"
+    bin.install_symlink "#{libexec}/bin/#{osname}_launcher" => "xtc"
+    bin.install_symlink "#{libexec}/bin/#{osname}_launcher" => "xam"
```

Comment:
```suggestion
    %W[xec xtc xam].each do |bin_name|
      (bin/bin_name).write_env_script libexec/"bin/#{osname}_launcher",  Language::Java.overridable_java_home_env
    end
```

Reply:
OK, I've looked up the doc and read the source for [write_env_script](https://rubydoc.brew.sh/Pathname). Is it expected that OpenJDK will be installed (as required by the formula), but somehow the "java" executable is expected to not be in the path? I'm trying to understand both the anticipated environment, and the reasoning. Just to explain what that launcher program does: it picks up the command name from the name of the executable (or link) used, and then builds a command line to Java and executes it, so using a bash script in lieu of the link would likely break that.

Reply:
Unless this tool does something non-standard Java programs generally look for `JAVA_HOME` to base their command on, which won't be available unless we wrap the script with write_env_script.

Reply:
Does that mean that JAVA_HOME is not available to the e.g. `macos_launcher`, unless we explicitly set it up by creating a shell script? That seems to be what you are saying, but I want to make sure that I understand.

Reply:
There is no way to guarantee the user has `JAVA_HOME` set for the formula to use, unless we set it in a shell script wrapper.

Reply:
We can use a config file (same name as the link file + .cfg) with a pointer to the Java executable. Would something like this work to create those files? (I am a newbee with ruby.) 

    %W[xec xtc xam].each do |bin_name|
        bin.install_symlink "#{libexec}/bin/#{osname}_launcher" => "#{bin_name}"
        (bin/"#{bin_name}.cfg").write "exec=#{Language::Java.overridable_java_home_env}/bin/java"
    end

Reply:
It would work technically to write a file but configuration files should be in `etc` so I would prefer the `write_env_script` approach.

Reply:
Sean, would it be reasonable to test from the install (from within the Formula) for the presence of `java` in the path? If it is present (as it generally is, when openjdk is installed), the link files will work as-is.

In the case when `java` is absent, I could emit the necessary xec.cfg (etc.) file with the `Language::Java.overridable_java_home_env` value incorporated into it (as above).

If this is reasonable, would you suggest that I use `shell_output` to test for the ability to execute `java`? Otherwise, I could get `ENV['PATH']` and do a `.split(File::PATH_SEPARATOR)` and check each directory, but I assume there's already something like that in a Homebrew library (although I haven't found one).

These must all seem like idiotic questions, so thank you in advance for your patience in explaining.

Reply:
So, what's wrong with using the wrapper I suggested? Almost all Java software in homebrew uses this pattern, I'm not sure why it wouldn't work here.